app/build.gradle

```diff
@@ -176,6 +179,12 @@ dexcount {
 }
 */
 
+unMock {
```

Comment:
Did you need that for the unit tests? Would be nice to have more comments :)

Reply:
> @Helium314 If you are happy with it we can merge that to main.
> Awesome work!

Thanks!
There are some small things that should be addressed (will do later today if I have the time):
* @jamal2362 found some weird CORS error that stops loading the page on browser start, even if there are no blocklists loaded (which I really don't understand). Simply reloading the page works. I found a similar stop of page loading once, but am unable to reproduce.
https://github.com/Slion/Fulguris/blob/f114d006b0c5904ab4f2d4bbfb37c8c46488c252/app/src/main/java/acr/browser/lightning/adblock/AbpBlocker.kt#L268
Changing `request.isForMainFrame` to `pageUrl == ""` stops the error, but also leads to wrong third party detection when switching page (and can result in wrong blocks). Since I can't reliably reproduce the error, I find it hard to fix.
* Auto-update frequency is only a lower limit. A blocklist can specify an _expire_ time, usually a few days, and when checking for auto-updates the longer of auto-update frequency and expire is used. Since the lack of daily updates might confuse the user, I need to add some excplanation. I also thought about ignoring the expire time, but don't want to put unnecessary load on the blocklist servers.
* _update now_ button exists in two menus, need to decide from which menu to remove it

> > Currently the built-in blocklist is completely independent from all others, even though it actually is a copy of easylist. This could be changed so that it is used as starting point, until it is updated for the first time. Then actual (more recent) easylist would be used instead.
> 
> Looks like you tackled that in later changes.

Actually I did not... so the choice remains:
use a full built-in list that probably should be updated on every browser update
use a minimal built-in list that gets replaced by a different one (probably easylist) after the first list auto-update

> > Code for using malware lists is implemented, but there is no actual malware list to be used. Malware checks are done before adblock checks, so any exclusion in adblock-list can't stop malware list from working.
> > Should we add some list, e.g. from https://gitlab.com/curben/urlhaus-filter#hosts-based? My current idea would be to not have the list adjustable by user, only have some enable/disable switch.
>
> Are malware and adblock list handled the same? Do they just provide like different level of content blocking?

They are handled the same, but it is possible to have exclusions on the blocklists. If for some reason a malware page is on the exclusion list, it will not be blocked.
For this reason, malware blocking happens before checking for exclusions, and making this filter container user accessible would require additional work...
If the user doesn't use any lists with weird exclusions, malware blocking will work the same if the user adds a malware list like a normal blocklist.
(I plan to later add more filter rules that are used in AdGuard and [uBlockOrigin](https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#extended-syntax), which includes an `important` rule that can't be overridden by exclusion. `important` rules would go into the (maybe renamed) malware list, and basically give the user the possibility to have a reliable malware list.)

> Disabling ad-block in the settings should disable all sorts of blocking yes. Though it should really be renamed content blocker at some point.

Alright, will finally rename it to content blocker :)